controllers/dashboard_test.go

```diff
@@ -0,0 +1,38 @@
+package controllers_test
+
+import (
+	"net/http"
+
+	. "go-crawler-challenge/tests"
+	. "go-crawler-challenge/tests/fixtures"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("DashboardController", func() {
+	AfterEach(func() {
+		TruncateTable("user")
+	})
+
+	Describe("GET /", func() {
+		Context("when the user has NOT signed in yet", func() {
+			It("redirects to sign-in page", func() {
+				response := MakeRequest("GET", "/dashboard", nil)
+				currentPath := GetCurrentPath(response)
+
+				Expect(response.StatusCode).To(Equal(http.StatusFound))
+				Expect(currentPath).To(Equal("/user/sign_in"))
+			})
+		})
+
+		Context("when the user has already signed in", func() {
```

Comment:
In terms of ordering of tests, usually, the happy path is listed before the sad paths 💭

Reply:
Uudated here https://github.com/Lahphim/go-crawler-challenge/pull/51/commits/ec0ed00ac14f5ca9bcd6cde4d8b644d5f287c5d0